openedx_events/learning/data.py

```diff
@@ -296,3 +296,23 @@ class ProgramCertificateData:
     status = attr.ib(type=str)
     url = attr.ib(type=str)
     certificate_available_date = attr.ib(type=datetime, default=None)
+
+
+@attr.s(frozen=True)
+class ExamAttemptData:
+    """
+    Attributes defined for the Open edX Exam downstream effects.
+
+    Arguments:
+        user_id (int): identifier of the user to which the grade override belongs
+        course_key (CourseKey): identifier of the course to which the grade override belongs
+        usage_key (UsageKey): identifier of the content that will get a grade override
+        requesting_user (UserData): user triggering event, sent only when instructor modifies an exam attempt
+        credit_requirement_status (str): status to change the student's CreditRequirement to
+    """
+
+    user_id = attr.ib(type=int)
```

Comment:
I noticed @MichaelRoytman's PR uses the UserData object here. Do you think we should use that?

Reply:
Looking at the parameters of the [downstream functions](https://2u-internal.atlassian.net/wiki/spaces/PT/pages/577699957/M6+Hackathon+Outline#(Old)-Parameters-of-downstream-functions-in-edx-platform%3A), it looks like most of them take user_id as a parameter anyways, so I think user_id is the way to go here.

Also I believe Michael's PR just has placeholder data for now.

Reply:
I think it would make sense to use the `UserData`. It seems to be more common to use that than a `user_id`. But I do see uses of both, so there's precedence to use both.

I don't think we have to model our events off the way they're being consumed now. Ideally, they shouldn't be so context aware. I think, for me, it's less, "the Instructor Service currently only expects a `user_id` so that's what we should put in the event" and more "how can I best and most generically represent the concept of a user in my event"?

I guess the only consideration is maybe sending PII when we don't need to?

Reply:
I think there's a good chance future downstream logic would use the pii. This probably gets hooked up to name affirmation at some point.